packages/app/src/vis-packs/core/complex/MappedComplexVis.tsx

```diff
@@ -49,12 +59,16 @@ function MappedComplexVis(props: Props) {
 
   const [mappedArray] = useMappedArray(value, slicedDims, slicedMapping);
 
-  const {
-    phaseArray,
-    phaseDomain = DEFAULT_DOMAIN,
-    amplitudeArray,
-    amplitudeDomain = DEFAULT_DOMAIN,
-  } = usePhaseAmplitudeArrays(mappedArray, scaleType);
+  const { phaseValues, phaseBounds, amplitudeValues, amplitudeBounds } =
+    useMemo(() => getPhaseAmplitudeValues(mappedArray.data), [mappedArray]);
+
+  const phaseArray = useBaseArray(phaseValues, mappedArray.shape);
+  const amplitudeArray = useBaseArray(amplitudeValues, mappedArray.shape);
+
+  const phaseDomain =
+    useValidDomainForScale(phaseBounds, scaleType) || DEFAULT_DOMAIN;
+  const amplitudeDomain =
+    useValidDomainForScale(amplitudeBounds, scaleType) || DEFAULT_DOMAIN;
```

Comment:
To solve a shadowing issue, I decided to refactor `usePhaseAmplitudeArrays` with some other hooks that we have. In the end, it was short enough to inline in the component. I didn't use `usePhaseAmplitudeValues` based on `createMemo`, because I preferred to pass `mappedArray` instead of `mappedArray.data` in the dependencies array. After that, since `usePhaseAmplitudeValues` was only used in one place and alone in a `hooks` file, I decided to remove it altogether (see `MappedComplexeLineVis` above)